axelor-base/src/main/java/com/axelor/apps/base/web/PartnerApiConfigurationController.java

```diff
@@ -26,11 +31,32 @@ public void fetchData(ActionRequest request, ActionResponse response) throws Axe
     PartnerApiConfiguration partnerApiConfiguration =
         Beans.get(PartnerApiConfigurationRepository.class).find(apiConfigId);
 
+    Object partnerId = request.getContext().get("_id");
+    Partner partner;
+    if (partnerId != null) {
+      partner = JPA.find(Partner.class, Long.parseLong(partnerId.toString()));
+    } else {
+      partner = new Partner();
+    }
+
     String result = Beans.get(PartnerApiFetchService.class).fetch(partnerApiConfiguration, siret);
 
     if (!StringUtils.isEmpty(result)) {
-      response.setInfo(result);
-      response.setCanClose(true);
+      Beans.get(PartnerApiCreateServiceImpl.class).setData(partner, result);
```

Comment:
* The interface should be injected, not the implementation.
* the controller should call a single method. so `PartnerApiCreateService` must have `PartnerApiFetchService` as a dependency.
* `PartnerApiCreateService` and the method `setData` should probably be renamed.